lib/instrumentation/span.js

```diff
@@ -112,22 +141,10 @@ Span.prototype._encode = function (cb) {
   if (!this.ended) return cb(new Error('cannot encode un-ended span'))
 
   if (this._agent._conf.captureSpanStackTraces) {
-    // TODO: This is expensive! Consider if there's a way to cache some of this
-    stackman.callsites(this._stackObj, function (err, callsites) {
-      if (!callsites) {
-        debug('could not capture stack trace for span %o', {id: self.transaction.id, name: self.name, type: self.type, err: err && err.message})
-        done()
-        return
-      }
-
-      if (!process.env.ELASTIC_APM_TEST) callsites = callsites.filter(filterCallsite)
-
-      var next = afterAll(done)
-
-      callsites.forEach(function (callsite) {
-        parsers.parseCallsite(callsite, false, self._agent._conf, next())
-      })
-    })
+    this._stackObj.then(
```

Comment:
Today `this._stackObj` will be `null` if `captureSpanStackTraces` is `false`, in which case this is safe. But as this is a pretty loose relationship, I wonder if we should add a guard to ensure we can't get to here unless `this._stackObj` is set. Just to make it more future-proof. What do you think?

Reply:
It might just be me that's overly cautious 😅

Reply:
Sounds reasonable.

Reply:
fixed